Content.Server/PowerSink/PowerSinkComponent.cs

```diff
@@ -1,8 +1,36 @@
-﻿namespace Content.Server.PowerSink
+﻿using Robust.Shared.Audio;
+
+namespace Content.Server.PowerSink
 {
     /// <summary>
     /// Absorbs power up to its capacity when anchored then explodes.
     /// </summary>
     [RegisterComponent]
-    public sealed class PowerSinkComponent : Component {}
+    public sealed class PowerSinkComponent : Component
+    {
+        /// <summary>
+        /// When the power sink is nearing its explosion, warn the crew so they can look for it
+        /// (if they're not already).
+        /// </summary>
+        [DataField("sentImminentExplosionWarning")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public bool SentImminentExplosionWarningMessage = false;
+
+        /// <summary>
+        /// If explosion has been triggered, time at which to explode.
+        /// </summary>
+        [DataField("explosionTime")]
+        public System.TimeSpan? ExplosionTime = null;
```

Comment:
Okay I double-checked and nullable timespans should work with TimeOffsetSerializer so please use that for customtypeserializer.

The reason we use that is because it will offset the explosion time against the game time so if the map gets saved and loaded (e.g. for persistence) then despite the new timing the offset is still correct.